File: algorithms/Find and Replace Pattern/Find and Replace Pattern.java
class Solution {
    public List<String> findAndReplacePattern(String[] words, String pattern) {
        List<String> result=new ArrayList<>();
        for(String word:words) {
            Map<Character,Character> map=new HashMap<>();
            Set<Character> set=new HashSet<>();
            int i=0;
            for(;i<word.length();i++) {
                char ch=pattern.charAt(i);
                if(map.get(ch)==null) {
                    if(set.contains(word.charAt(i))) break;
                   map.put(ch, word.charAt(i));
                    set.add(word.charAt(i));
                }
                else {
                    char mc=map.get(ch);
                    if(mc!=word.charAt(i)) break;
                }
            }
            if(i==pattern.length()) result.add(word);
        }
        return result;
    }
}

File: scripts/algorithms/F/Find and Replace Pattern/Find and Replace Pattern.cpp
			//😉If you Like the repository don't foget to star & fork the repository😉
class Solution {
public:
    
    vector<int> found_Pattern(string pattern)
    {
	   // if string is empty return empty vector.
        if(pattern.size() == 0)
            return {};
        
        vector<int> v;
		
		// ind variable for keeping track of unique characters
        int ind = 0;
		
        unordered_map<char,int> mp;
        for(int i = 0; i<pattern.size(); ++i)
        {
			// if character not present in map, insert the char with an index,
			// increment index because for next unique character the index should be differnt.
           if(mp.find(pattern[i]) == mp.end())
           {
               mp.insert({pattern[i],ind++});
			   // also  push the index to v(numeric  pattern vector)
			   // mp[pattern[i]] gives the key to that character, here key is ind(which we are giving to every unique character)
               v.push_back(mp[pattern[i]]);
           }
            else
            {
				// if char is already in map than push index mapped to that character into the vector
                v.push_back(mp[pattern[i]]);
            }
        }
		// return numeric pattern
        return v;
    }
    vector<string> findAndReplacePattern(vector<string>& words, string pattern) {
       
		// store numeric patten of Pattern string in v
        vector<int> v = found_Pattern(pattern);
        
        int n = words.size();
        vector<string> ans;
        
		// iterating and comparing pattern of each word with v(numeric pattern of Pattern)
        for(int i = 0; i<n; ++i)
        {
            vector<int> pattern_word = found_Pattern(words[i]);
            
			// if matched add words[i] to answer vector
            if(v == pattern_word)
                ans.push_back(words[i]);
        }
        
        return ans;
    }
};
